### PR TITLE
Add unit test for ThriftHiveMetastoreClient::alternativeCall

### PR DESCRIPTION
**Pull request by skrzypo987**
## Additional context and related issues
https://github.com/trinodb/trino/pull/15462

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestThriftHiveMetastoreClient.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        assertThat(connectionCount.get()).isEqualTo(2);
+        assertThat(chosenOption.get()).isEqualTo(1);
+
+        assertThat(client.alternativeCall(e -> false, chosenOption, failure, sucesss1, sucesss2))
+                .isEqualTo("first");
+        // Alternative call should use chosenOption
+        assertThat(connectionCount.get()).isEqualTo(2);
+    }
+
+    private static class TTransportMock
```
Comment (kokosing) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestThriftHiveMetastoreClient.java:72]: `com.google.common.reflect.Reflection#newProxy`?

Comment (skrzypo987) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestThriftHiveMetastoreClient.java:72]: I was under the impression that using mocking libraries is forbidden in Trino

Comment (kokosing) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestThriftHiveMetastoreClient.java:72]: It is not mocking library, but reflections.

Comment (skrzypo987) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestThriftHiveMetastoreClient.java:72]: Reflection is one way to implement mocking. But if you believe it's fine, I'll use it. The less code the better.

Comment (kokosing) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestThriftHiveMetastoreClient.java:72]: Mocking is fine. See we have entire `MockConnector` :)

Comment (skrzypo987) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestThriftHiveMetastoreClient.java:72]: > `com.google.common.reflect.Reflection#newProxy`?

Not so fast:
`java.lang.IllegalArgumentException: class org.apache.thrift.transport.TTransport is not an interface`

**Comment by kokosing** [comment]
Thanks. Merged.

**Pull request merged**